runelite-client/src/main/java/net/runelite/client/plugins/microbot/util/bank/enums/BankLocation.java

```diff
@@ -99,19 +101,38 @@ public boolean hasRequirements() {
         switch (this){
             case CRAFTING_GUILD:
                 if (hasLineOfSight) return true;
-                return Microbot.getClient().getRealSkillLevel(Skill.CRAFTING) >= 40
+                return Rs2Player.getRealSkillLevel(Skill.CRAFTING) >= 40
                         && (Rs2Equipment.isWearing("brown apron") || Rs2Equipment.isWearing("golden apron"));
             case LUMBRIDGE_BASEMENT:
                 return Rs2Player.getQuestState(Quest.RECIPE_FOR_DISASTER__ANOTHER_COOKS_QUEST) == QuestState.FINISHED;
             case COOKS_GUILD:
                 if (hasLineOfSight && Rs2Player.isMember()) return true;
-                boolean hasVarrockHardDiary = Microbot.getClientThread().runOnClientThread(() -> Microbot.getClient().getVarbitValue(Varbits.DIARY_VARROCK_HARD) == 1);
-                boolean hasMaxedCooking = Microbot.getClient().getRealSkillLevel(Skill.COOKING) >= 99;
+                boolean hasVarrockHardDiary = Microbot.getVarbitValue(Varbits.DIARY_VARROCK_HARD)  == 1;
+                boolean hasMaxedCooking = Rs2Player.getRealSkillLevel(Skill.COOKING) >= 99;
                 boolean isWearingCooksGuild = Rs2Equipment.isWearing("chef's hat") ||
                         (Rs2Equipment.isWearing("cooking cape") || Rs2Equipment.isWearing("cooking hood")) ||
                         (Rs2Equipment.isWearing("max cape") || Rs2Equipment.isWearing("max hood")) ||
                         (Rs2Equipment.isWearing("varrock armour 3") || Rs2Equipment.isWearing("varrock armour 4"));
                 return Rs2Player.isMember() && isWearingCooksGuild && (hasVarrockHardDiary || hasMaxedCooking);
+            case WARRIORS_GUILD:
+                if (hasLineOfSight) return true;
+                return (Rs2Player.getRealSkillLevel(Skill.ATTACK) >= 99 || Rs2Player.getRealSkillLevel(Skill.STRENGTH) >= 99) ||
```

Comment:
I think warriors guild is missing a Rs2Player.isMember check

Reply:
thats right, it is!

https://oldschool.runescape.wiki/w/Guilds

Reply:
The checks for Warrior's Guild are now there in the latest commit c54c442